### ISSUE TITLE
[Bug] Scraping: `excludeTags` is ignored when `includeTags` is given. (FIR-627)

### ISSUE THREAD

**Issue by wottpal** [mendableai/firecrawl#700]
**Describe the Bug**
When I'm scraping I can't make `excludeTags` (neither in `html` or `markdown` formats). I assume it's because I need `includeTags` and only one parameter (`includeTags` xor `excludeTags`) is supposed to be working. – Is that correct? 

I constantly have the use case that I want to **include** something but **exclude** certain child tags. Is that use case supported?

**Comment by mogery** [comment]
Hi there, both parameters combine should work too, I'll look into this.

**Comment by wottpal** [comment]
Great, any updates on this? :)

**Comment by deltacodepl** [comment]
> Great, any updates on this? :)

It Does not work, it gives full includeTags part

**Comment by herzner** [comment]
True, the includeTags takes precedence over excludeTags. I am also facing the issue for the same use case (include certain tags but get rid of some unwanted childs).

I checked the code ([removeUnwantedElements.ts](https://github.com/mendableai/firecrawl/blob/main/apps/api/src/scraper/scrapeURL/lib/removeUnwantedElements.ts)):

```
  if (
    scrapeOptions.includeTags &&
    scrapeOptions.includeTags.filter((x) => x.trim().length !== 0).length > 0
  ) {
    // Create a new root element to hold the tags to keep
    const newRoot = load("<div></div>")("div");
    scrapeOptions.includeTags.forEach((tag) => {
      soup(tag).each((_, element) => {
        newRoot.append(soup(element).clone());
      });
    });
   // -> there is the problem :-)
    return newRoot.html() ?? "";
  }
  
  // handle the excludeTags ...
```

Instead of handling the excludeTags, the html for the new root is returned.
This could be a fix:

`soup.root().empty().append(newRoot);`

**Comment by CyrilShch** [comment]
Any updates on this? :)
Is it planned to be fixed? Thx
